### PR TITLE
Give a mission to the drone when the mission is launched, and end the mission when the drone is back home

### PR DESCRIPTION
**Pull request by Anghul**

### CODE REVIEW

**Review by GabrielFleischer — APPROVED** [review]
I worked on the PR, so I accept it

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/service/drone/DroneExecutorTest.kt [hunk 1]
```diff
@@ -56,7 +57,7 @@ class DroneExecutorTest {
 
         val droneData = DroneDataImpl(droneService)
 
-        val executor = DroneExecutorImpl(droneService, locationService, droneData)
+        val executor: DroneExecutor = DroneExecutorImpl(droneService, locationService, droneData)
```
Comment (jonasblanc) [inline app/src/androidTest/java/ch/epfl/sdp/drone3d/service/drone/DroneExecutorTest.kt:60]: General comment for the file, you could init the services in one place instead of in every test

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/MainActivityTest.kt [hunk 2]
```diff
@@ -94,6 +94,7 @@ class MainActivityTest {
         onView(withId(R.id.log_in_button))
                 .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
         onView(withId(R.id.log_in_button)).perform(click())
+
```
Comment (jonasblanc) [inline app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/MainActivityTest.kt:97]: Try to avoid modifying files that are not related to the PR

**Review by jonasblanc — APPROVED** [review]

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/service/drone/DroneExecutorTest.kt (hunk 1 above)
Comment (GabrielFleischer) [reply app/src/androidTest/java/ch/epfl/sdp/drone3d/service/drone/DroneExecutorTest.kt:60]: As I'm mocking it, so changing its state, I don't want to have mocked data from older tests making it pass. Because then the order in which the tests are run would have an impact on them passing.
